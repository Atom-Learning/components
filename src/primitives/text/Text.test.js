import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import React from 'react'

import { Text } from '.'

describe(`Text component`, () => {
  it('renders some text', async () => {
    const { container } = render(<Text>TEXT</Text>)

    await screen.getByText('TEXT')
    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    render(
      <main>
        <Text>TEXT</Text>
      </main>,
      document.body
    )

    const results = await axe(document.body)
    expect(results).toHaveNoViolations()
  })
})
