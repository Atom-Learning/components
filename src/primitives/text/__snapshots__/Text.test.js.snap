// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component renders some text 1`] = `
._iWlAZF {
  max-width: 100ch;
}

._heQxlO {
  font-weight: 400;
}

._dFQJLr {
  color: var(--colors-tonal900);
}

._oIZSq {
  font-family: var(--fonts-sans);
}

._laGdYr {
  line-height: 1.4;
}

._bIcFyi {
  letter-spacing: 0.02em;
}

._dIHnma {
  font-size: var(--fontSizes-md);
}

<div>
  <p
    class="_dIHnma _bIcFyi _laGdYr _oIZSq _dFQJLr _heQxlO _iWlAZF scid-haMvEs"
  >
    TEXT
  </p>
</div>
`;
